docs: Add a workflow example for publishing Universal macOS app with signing certificate

8beeeaaat opened the pull request:
This workflow ensures that the app is properly signed and ready for distribution on macOS.
Users can start using the app right away since the signed .dmg has been released.  (Users can skip the hassle of allowing the app to run in "Privacy & Security" after downloading it. )

- sample: my CD result
https://github.com/8beeeaaat/tauri_distribution_sample/releases

This workflow creates or updates a GitHub release, builds the app, and uploads the artifacts to the release on each push to the `release` branch.
It includes steps for setting up the environment, importing the Apple Developer Certificate, and building and publishing the app using tauri-action.

- Imports the Apple Developer Certificate.
- Verifies the imported certificate.
- Prevents keychain from locking automatically for 3600 seconds. https://github.com/tauri-apps/tauri-action/issues/941

rdlabo commented:
This is great! I had to worry for over half a day without run `security set-keychain-settings -t 3600 -u build.keychain`. Once this is maintained in the documentation, no one will have any trouble!

FabianLars commented:
Thank you! just one minor request. could we remove the whole TAG_NAME stuff and let tauri-action handle this (like in https://github.com/tauri-apps/tauri-action/blob/dev/examples/publish-to-auto-release.yml#L58 )? Just so that the example better focuses on the certificate stuff.

8beeeaaat commented:
@rdlabo @FabianLars 

Thank you for your reviews!
I've addressed your comments and made the necessary changes. Please let me know if there's anything else that needs improvement.
